builder/dockerignore.go

```diff
@@ -26,8 +26,8 @@ type DockerIgnoreContext struct {
 //
 // TODO: Don't require a ModifiableContext (use Context instead) and don't remove
 // files, instead handle a list of files to be excluded from the context.
-func (c DockerIgnoreContext) Process(filesToRemove []string) error {
-	f, err := c.Open(".dockerignore")
+func (c DockerIgnoreContext) Process(ignoreFile string, filesToRemove []string) error {
+	f, err := c.Open(ignoreFile)
 	// Note that a missing .dockerignore file isn't treated as an error
```

Comment:
I'm not sure this comment/logic is true in the case of someone being explicit about it.  I would prefer if we threw an error if we were told to use a file and that file didn't exist.  I think it should only silently ignore the "not found" error in the default case.